### PR TITLE
Add numpy 2.0 compatibility

### PR DESCRIPTION
**Pull request by zm711**
this is built off of #232 

Since I don't know array wrapping perfectly I may need to tweak this again.

### CODE REVIEW

**Comment by zm711** [comment]
So my fix, fixes old numpy but breaks new. Let me try again.

**Comment by zm711** [comment]
@apdavison, both pre and post NumPy 2.0 tests are now passing on this PR.

**Comment by axelande** [comment]
Any plans of including this PR and release a new version that is compatible with numpy2?
